### DISCUSSION TITLE
Is it possible to only hydrate part of the page?

### DISCUSSION THREAD

**Discussion by ScottPierce** [vikejs/vike#1009] [Discussion] [answered]
I understand that we can have client only files with `.page.client.tsx`, and server side only files with `page.server.tsx`, and that using `.page.tsx` will pre-render on the server, and then hydrate in the client.

Is it possible to only hydate a small part of the page? i.e. if I have a large page that is 95% static, with a small part of the page that requires some dynamic interactivity with react, is it possible to only hydrate that small portion, or do we have to hydrate the entire page?

Is the performance cost of hydrating something we shouldn't be concerned with?

Astro provided something similar to this [called Islands](https://docs.astro.build/en/concepts/islands/), which we thought was pretty neat.

**Comment by brillout** [comment, marked as answer]
Yes that's possible. (FYI `https://vite-plugin-ssr.com` doesn't even hydrate anything.)

But note that you'll be complexifying your architecture. So that's something to be aware of.

In general, my recommendation is to go with the beaten path (full page hydration) and optimize only if you aren't happy with performance.

(FYI, while the flexibility of `vite-plugin-ssr` can be quite neat, it can also leave too much freedom for users to make mistakes. The upcoming [`vike-react`](https://github.com/brillout/vike-react/tree/main/vike-react) integration implements the classic approach hydrating the full page.)

Now that you're warned, if you want to have fun implementing an island architecture then, yes, that's possbile.

**Reply by ScottPierce** [reply]
I realized after posting this that we actually do the hydration ourselves, and that it's in the `_default.page.client.tsx`. Basically, what you are saying is that we can just choose to call `hydrateRoot` on whatever we want.

That's very powerful for our use case. Thank you.

**Reply by brillout** [reply]
Exactly.
